contracts/src/protocol/IPTokenStaking.sol

```diff
@@ -320,231 +327,274 @@ contract IPTokenStaking is IIPTokenStaking, Ownable2StepUpgradeable, ReentrancyG
-
+        if (stakeAmount < minStakeAmount) {
+            revert Errors.IPTokenStaking__StakeAmountUnderMin();
+        }
+        if (commissionRate < minCommissionRate) {
+            revert Errors.IPTokenStaking__CommissionRateUnderMin();
+        }
+        if (commissionRate > maxCommissionRate) {
+            revert Errors.IPTokenStaking__CommissionRateOverMax();
+        }
+        payable(address(0)).transfer(stakeAmount);
+        emit CreateValidator(
+            validatorUncmpPubkey,
+            moniker,
+            stakeAmount,
+            commissionRate,
+            maxCommissionRate,
+            maxCommissionChangeRate,
+            supportsUnlocked ? 1 : 0,
+            msg.sender,
+            data
+        );
         _refundRemainder(remainder);
-
-        emit CreateValidator({
-            validatorUncmpPubkey: validatorUncmpPubkey,
-            validatorCmpPubkey: validatorCmpPubkey,
-            moniker: vm.moniker,
-            stakeAmount: stakeAmount,
-            commissionRate: vm.commissionRate,
-            maxCommissionRate: vm.maxCommissionRate,
-            maxCommissionChangeRate: vm.maxCommissionChangeRate
-        });
     }
 
-    /// @notice Entry point for staking IP token to stake to the given validator. The consensus chain is notified of
+    // TODO: update validator method (next version)
+
+    /*//////////////////////////////////////////////////////////////////////////
+    //                             Token Staking                              //
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Entry point to stake (delegate) to the given validator. The consensus client (CL) is notified of
     /// the deposit and manages the stake accounting and validator onboarding. Payer must be the delegator.
-    /// @dev When staking, consider it as BURNING. Unstaking (withdrawal) will trigger native minting.
+    /// @dev Staking burns tokens in Execution Layer (EL). Unstaking (withdrawal) will trigger minting through
+    /// withdrawal queue.
+    /// This method will revert if delegatorUncmpPubkey is not the sender of the validator.
     /// @param delegatorUncmpPubkey Delegator's 65 bytes uncompressed secp256k1 public key.
-    /// @param validatorCmpPubkey Validator's 33 bytes compressed secp256k1 public key.
+    /// @param validatorUncmpPubkey Validator's65 bytes uncompressed secp256k1 public key.
+    /// @param stakingPeriod The staking period.
+    /// @param data Additional data for the stake.
+    /// @return delegationId The delegation ID, always 0 for flexible staking.
     function stake(
         bytes calldata delegatorUncmpPubkey,
-        bytes calldata validatorCmpPubkey
+        bytes calldata validatorUncmpPubkey,
+        IIPTokenStaking.StakingPeriod stakingPeriod,
+        bytes calldata data
     )
         external
         payable
+        override
```

Comment:
The override modifier might not be necessary here, since this contract doesn’t inherit from any other contract.